subsys/bootloader/cmake/sign.cmake

```diff
@@ -119,14 +119,14 @@ foreach (slot ${slots})
       > ${signature_file}
       )
   elseif (CONFIG_SB_SIGNING_CUSTOM)
-    set(sign_cmd
-      ${CONFIG_SB_SIGNING_COMMAND}
-      ${hash_file}
-      > ${signature_file}
-      )
-    if (sign_cmd STREQUAL "" OR NOT EXISTS ${SIGNATURE_PUBLIC_KEY_FILE})
-      message(WARNING "You must specify a signing command and valid public key file.")
+    set(custom_sign_cmd "${CONFIG_SB_SIGNING_COMMAND}")
+
+    if ((custom_sign_cmd STREQUAL "") OR (NOT EXISTS ${SIGNATURE_PUBLIC_KEY_FILE}))
+      message(FATAL_ERROR "You must specify a signing command and valid public key file for custom signing.")
     endif()
+
+    string(APPEND custom_sign_cmd " ${hash_file} > ${signature_file}")
+    string(REPLACE " " ";" sign_cmd ${custom_sign_cmd})
```

Comment:
how do you know that a custom sign command outputs to standard out ?
User could have a command that did:
```
<command> <file> -o <signature file>
```

Reply:
The [Kconfig documentation](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/kconfig/CONFIG_SB_SIGNING_COMMAND.html?highlight=config_sb_signing_command#cmdoption-arg-CONFIG_SB_SIGNING_COMMAND) specifies this as a requirement for a custom command.

Reply:
then this is something to be improved it the bootloader / signing improvement / aligning tasks we must do.

Reply:
In future we should allow more flexibility here, maybe similar to how toolchains are handled.
Kconfig is not well suited for this purpose.

Reply:
Opened NCSDK-7814 to track and address this in future work.